addons/html_builder/static/src/core/building_blocks/basic_many2many.js

```diff
@@ -0,0 +1,84 @@
+import { Component, useRef, useState, onWillStart, onWillUpdateProps } from "@odoo/owl";
+import { useService, useAutofocus } from "@web/core/utils/hooks";
+import { debounce } from "@web/core/utils/timing";
+import { basicContainerBuilderComponentProps } from "./utils";
+import { DropdownItem } from "@web/core/dropdown/dropdown_item";
+import { Dropdown } from "@web/core/dropdown/dropdown";
+
+export class BasicMany2Many extends Component {
+    static template = "html_builder.BasicMany2Many";
+    static props = {
+        ...basicContainerBuilderComponentProps,
+        model: String,
+        fields: { type: Array, element: String, optional: true },
+        domain: { type: Array, optional: true },
+        limit: { type: Number, optional: true },
+        // createAction: { type: String, optional: true },
+        selection: { type: Array, element: Object },
+        setSelection: Function,
+        canCreate: { type: Boolean, optional: true },
+    };
+    static defaultProps = {
+        fields: [],
+        domain: [],
+        limit: 10,
+        canCreate: false,
+    };
+    static components = { Dropdown, DropdownItem };
+
+    setup() {
+        this.orm = useService("orm");
+        this.createInputRef = useRef("createInput");
+        this.state = useState({
+            selection: [],
+            searchResults: [],
+        });
+        this.onSearch = debounce(this.search.bind(this), 300);
+        // TODO focus on open dropdown, does not seem to work
+        useAutofocus();
+        onWillStart(async () => {
+            await this.handleProps(this.props);
+        });
+        onWillUpdateProps(async (newProps) => {
+            await this.handleProps(newProps);
+        });
+    }
+    async handleProps(props) {
+        this.state.searchResults = [];
+    }
+    search(ev) {
+        this._search(ev.target.value);
+    }
+    async _search(needle) {
+        const recTuples = await this.orm.call(this.props.model, "name_search", [], {
+            name: needle,
+            args: [].concat(Object.values(this.props.domain).filter((item) => item !== null)),
```

Comment:
why concat ? Object.values(this.props.domain).filter((item) => item !== null) isn't enough ?
Why do you need to remove item === null ? (How it s possible to have null item ? )

Reply:
Filter is kept from master - we do not know how broken the DOM of the page might be